src/toolbar.scss

```diff
@@ -143,47 +153,54 @@ $title-toolbar-height: 2.75rem;
   }
 
   &__overflow {
-    &.#{$fd-namespace}-popover {
-      .#{$block}__overflow__body {
-        @include fd-reset();
-
-        display: flex;
-        flex-direction: column;
-        max-width: 20rem;
-        padding: 0.1875rem 0.375rem;
-
-        > * {
-          display: block;
-          text-align: left;
-          width: auto;
-          margin: 0.1875rem 0;
-
-          @include fd-rtl() {
-            text-align: right;
-          }
-        }
+    @include fd-reset();
+    @include fd-flex(column);
+    @include fd-set-paddings-x-equal(0.375rem);
+    @include fd-set-paddings-y-equal(0.1875rem);
+
+    max-width: 20rem;
+    max-height: 50vh;
+    overflow: auto;
+
+    & .#{$block}__overflow-button {
+      justify-content: flex-start;
+      text-align: left;
+      width: auto;
+      margin: 0.1875rem 0;
+
+      @include fd-rtl() {
+        text-align: right;
+      }
 
-        > .#{$block}__separator {
-          height: 0.0625rem;
-          margin: 0.1875rem;
-          width: auto;
-          background-color: var(--sapToolbar_SeparatorColor);
+      &--menu {
+        @include fd-flex-center() {
+          justify-content: space-between;
         }
 
-        .#{$block}__overflow__form-label {
-          &.#{$fd-namespace}-form-label {
-            margin-top: 0.4375rem;
-          }
-        }
+        max-width: 100%;
+      }
+    }
 
-        .#{$block}__overflow__form-label,
-        .#{$block}__overflow__label {
-          overflow: hidden;
-          white-space: nowrap;
-          text-overflow: ellipsis;
-        }
+    > .#{$block}__separator {
+      @include fd-reset();
+
+      height: 0.0625rem;
+      margin: 0.1875rem;
+      width: auto;
+      background-color: var(--sapToolbar_SeparatorColor);
+    }
+
+    .#{$block}__overflow-form-label {
+      &.#{$fd-namespace}-form-label {
+        margin-top: 0.4375rem;
```

Comment:
Didn't we discuss multiple times (even in this PR) that you can't modify a component property inside another component? That's why we are creating modifier classes.

Reply:
copy paste error not with intention.

Reply:
During refactoring i have already removed many those kind of block, before submitting for review and kept those what you mentioned above to just to avoid the universal selector.